### DISCUSSION TITLE
Autotranslating Recipes

### DISCUSSION THREAD

**Discussion by ApfelBirneKreis** [mealie-recipes/mealie#3011] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Pls add an autotranslate feature of recipes.  We are a multilingual household and if everyone adds recipes in thei Mothertongue noone else can read it

### Please provide a concise description of the feature that would resolve your issue.

The recipes should be translated to the users language

### Please consider and list out some caveats or tradeoffs made in your design decision

no tradeoffs

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by RealFoxie** [comment]
How would this work well? Using some external api that can be set for mealie to use?

**Reply by ApfelBirneKreis** [reply]
correct. should be done within api calls or an internal translation engine

**Comment by Cruaier** [comment]
I'm looking for the same feature. I could image it could work in 2 ways:
1) allow storing the same recipe in multiple languages
2) add an external API for automated translation, or use something like https://libretranslate.com/ 

For external APIs, I could image a bring-your-own API key would be feasible.

**Comment by BhasherBEL** [comment]
Now that OpenAI integration is there, it would be possible to use it to automatically translate recipes.

**Comment by michael-genson** [comment]
The newest nightly has the ability to import via image using OpenAI: https://github.com/mealie-recipes/mealie/pull/3974; it can also optionally translate the recipe for you (with varying success depending on which model you use)

**Reply by Cruaier** [reply]
That's a nice feature but allowing to store the recipes in a different language or an "on-fly" translation would be more of what is suggested here.

**Comment by bf8392** [comment]
How can I translate receipes that are already in the database?
